src/main/java/com/mraof/minestuck/computer/editmode/EditToolDrag.java

```diff
@@ -0,0 +1,438 @@
+package com.mraof.minestuck.computer.editmode;
+
+import com.mojang.blaze3d.systems.RenderSystem;
+import com.mojang.blaze3d.vertex.BufferBuilder;
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.mojang.blaze3d.vertex.Tesselator;
+import com.mojang.blaze3d.vertex.VertexConsumer;
+import com.mojang.math.Matrix4f;
+import com.mraof.minestuck.Minestuck;
+import com.mraof.minestuck.block.machine.EditmodeDestroyable;
+import com.mraof.minestuck.block.machine.MachineBlock;
+import com.mraof.minestuck.entity.MSEntityTypes;
+import com.mraof.minestuck.entity.ServerCursorEntity;
+import com.mraof.minestuck.network.EditmodeFillPacket;
+import com.mraof.minestuck.network.MSPacketHandler;
+import com.mraof.minestuck.util.MSCapabilities;
+import net.minecraft.client.Camera;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.MultiBufferSource;
+import net.minecraft.client.renderer.RenderType;
+import net.minecraft.core.BlockPos;
+import net.minecraft.network.chat.Component;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.util.Mth;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.BlockItem;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.ClipContext;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.phys.AABB;
+import net.minecraft.world.phys.BlockHitResult;
+import net.minecraft.world.phys.Vec3;
+import net.minecraft.world.phys.shapes.Shapes;
+import net.minecraft.world.phys.shapes.VoxelShape;
+import net.minecraftforge.client.event.RenderLevelStageEvent;
+import net.minecraftforge.common.ForgeMod;
+import net.minecraftforge.event.TickEvent;
+
+import java.util.UUID;
+
+public class EditToolDrag
+{
+	
+	public static void doReviseCode(TickEvent.ClientTickEvent event)
+	{
+		Minecraft mc = Minecraft.getInstance();
+		if (mc.player == null || event.phase == TickEvent.Phase.END)
+			return;
+		
+		Player player = mc.player;
+		IEditTools cap = player.getCapability(MSCapabilities.EDIT_TOOLS_CAPABILITY, null).orElse(new EditTools());
```

Comment:
I imagine that it would be an error for the capability to be missing. I suggest throwing an error instead of silently ignoring the missing capability.